linode/manager
# upcoming: [M3-7940] - Update and cleanup Placement Group assign/unassign features

## Description 📝
Assigning/Unassigning linodes to/from placement groups has now been merged to alpha, warranting code updates to support the UI/UX for real API transactions (props/queries, loading states, invalidation).

While this touches a fair amount of files, we're looking at a `+493 −386` diff refactor aimed to simplify and optimize the feature code.

## Changes  🔄
- Better handling of props VS queries - case per case (PG features use path based modal/drawers - aka you can link to those) so the code to handle both case had to be touch in a number of files
- Modify the `RemovableSelectionsList` component to handle API request better (disabled and loading states). These updates include new non breaking, optional `false` defaulted props
- Removal of the `usePlacementGroupData` hook. It was an early over-optimization on my end from early development considerations, resulting in over-fetching and over-complicated loading states.
- Fix permissions
- Improve linode invalidation on assigning/unassigning
- Improve loading states for delete modal & Edit and Assign drawers
- Update tests and improve coverage
- Update some Naming conventions

## Preview 📷
https://github.com/linode/manager/assets/130582365/7032ee13-a18d-40c4-8c7f-0d1d486275fc

## How to test 🧪

⚠️ **Important Note:**
We're getting **gradual** alpha API implementation. This means some features are not yet functional. Here is what is not implemented as of this PR:
- regions limits (you will see `unknown` for limit values (ex: linodes column, "`2` out of `unknown`")
- linode POST, DELETE endpoints. You will not see PG assignment working while creating a linode, not will a linode be unassigned when deleted

### Prerequisites
- Have the dev admin `placement-group` customer tag
- Switch to the dev API
- Have the "Placement Group" feature flag on
- Have MSW off

### Verification steps
- Navigate to http://localhost:3000/placement-groups and play around with
  - Create Drawer
  - Editing drawer
  - Delete Modal
- Navigate to http://localhost:3000/placement-groups/{id}/linodes and play around with
  - Assign Linode drawer
  - Unassign Modal

When testing both, navigate back and forth to confirm query invalidations and that the data is updated according to your latest actions

## As an Author I have considered 🤔

*Check all that apply*
